### PR TITLE
feature/microsoft_365_defender_add_urlchain_field

### PR DESCRIPTION
**Pull request by clement-burtscher-sekoia**
## Description

Relates to https://github.com/SekoiaLab/integration/issues/1858

### Added

- Parse new custom fields:
    - `microsoft.defender.url_chain`

## Type of Change

- [ ] New intake format
- [x] Update to existing intake format
- [ ] Bug fix
- [ ] Documentation update
- [ ] Other (please describe):

## Affected Intake Formats

- Vendor/product: Microsoft/microsoft-365-defender

### 🔒 CRITICAL - Security Requirements (MUST BE VERIFIED FIRST)

- [ ] **No real email addresses** in test files (use: user@example.com, test@example.org)
- [ ] **No real passwords or API keys** (use: "P@ssw0rd!", "xxxxxxxxxxxx", "FAKE_TOKEN_123")
- [ ] **No real IP addresses** (use TEST-NET ranges: 192.0.2.x, 198.51.100.x, 203.0.113.x)
- [ ] **No real phone numbers** (use: +1-555-0100 or similar test numbers)
- [ ] **No real names or PII** (use: "John Doe", "Jane Smith", "User123")
- [ ] **No real company/organization names** in test data (unless public vendors)
- [ ] **No real usernames, hostnames, or domain names** (use: user1, host.example.com, test.corp)
- [ ] All test data has been anonymized and verified

### Required for All PRs

- [ ] Code has been linted
    - [ ] with the linter for manifest, smart-descriptions, and test files (`uv run python linter.py check --changes` in utils/)
    - [ ] with Prettier for the parser code files (`npx prettier --write <module>/<format>/ingest/parser.yml`)
- [ ] All CI/CD checks pass
- [ ] Changes follow the contribution guidelines in `CONTRIBUTING.md`

### Required for New/Updated Intake Formats

- [ ] Clear descriptions provided for modules and formats
- [ ] Logo files included for new modules/formats
- [ ] Parser test coverage is at least 75%
- [ ] At least one smart-description is provided
- [ ] README.md updated (if applicable)
- [ ] Taxonomy mappings are correct and documented
- [ ] Test cases cover edge cases and error handling

### Testing

- [ ] Local tests pass (`uv run pytest` in utils/)
- [ ] Sample data is representative and **completely anonymized**
- [ ] **No sensitive information** (PII, credentials, real IPs, real emails) in test files
- [ ] Test data uses example.com/example.org domains and TEST-NET IP ranges
- [ ] Parser handles malformed data gracefully

### Documentation

- [ ] Code changes are documented
- [ ] Smart-descriptions are clear and informative
- [ ] Field mappings are explained

## Related Issues

Closes #

## Summary by Sourcery

New Features:
- Introduce the microsoft.defender.url_chain custom field for URL click events.

### CODE REVIEW

File: Microsoft/microsoft-365-defender/ingest/parser.yml [hunk 1]
```diff
@@ -880,6 +880,7 @@ stages:
           event.dataset: "url_click_events"
           event.category: ["web"]
           event.type: ["info"]
+          microsoft.defender.url_chain: "{{json_event.message.properties.UrlChain | from_json}}"
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Microsoft/microsoft-365-defender/ingest/parser.yml:887]: Fixed with commit [Apply changes requested by Copilot review](https://github.com/SEKOIA-IO/intake-formats/pull/2772/changes/dc15a57135f0f133ecdad36249cc32fef6c7562a)

File: Microsoft/microsoft-365-defender/tests/test_url_click_events.json [hunk 2]
```diff
@@ -28,7 +28,11 @@
       "defender": {
         "report": {
           "id": "33333333-3333-3333-3333-333333333333"
-        }
+        },
+        "url_chain": [
+          "https://example.com/index.php?q=some+stuff",
+          "https://example.com/index.php?q=some+stuff&p=2"
+        ]
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Microsoft/microsoft-365-defender/tests/test_url_click_events.json:35]: Fixed with commit [Apply changes requested by Copilot review](https://github.com/SEKOIA-IO/intake-formats/pull/2772/changes/dc15a57135f0f133ecdad36249cc32fef6c7562a)

**Review by mchupeau-sk — APPROVED** [review]
LGFM
